File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool comp(string &a,string &b){
        return a+b>b+a;
    }
    string largestNumber(vector<int>& nums) {

       vector<string>s;
        for(int i:nums){
            s.push_back(to_string(i));
            
        }
        sort(s.begin(), s.end(),comp);  
        string a="";
        int t=0;
        for(string i:s){
            a=a+i;
            if(i!="0"){
                t=1;
            }
        }
        if(t==1){
            return a;
        }
        return "0";
    }
};

File: 0179-largest-number/0179-largest-number.py
from functools import cmp_to_key
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        

        arr = [3, 30, 34, 5, 9]
        s=""
        def compare(a, b):
            if str(a) + str(b) > str(b) + str(a):
                return -1
            elif str(a) + str(b) < str(b) + str(a):
                return 1
            return 0

        nums.sort(key=cmp_to_key(compare))

        for i in nums:
             s=s+str(i)
        if s[0]=="0":
            return "0"
        return s